turbopack/crates/turbopack-core/src/module_graph/mod.rs

```diff
@@ -1352,17 +1381,26 @@ impl Visit<SingleModuleGraphBuilderNode> for SingleModuleGraphBuilder<'_> {
                     // .await?;
 
                     refs.iter()
-                        .flat_map(|(ty, modules)| modules.iter().map(|m| (ty.clone(), *m)))
-                        .map(async |(ty, target)| {
+                        .flat_map(|(ty, export, modules)| {
+                            modules.iter().map(|m| (ty.clone(), export.clone(), *m))
+                        })
+                        .map(async |(ty, export, target)| {
                             let to = if ty == COMMON_CHUNKING_TYPE {
                                 if let Some(idx) = visited_modules.get(&target) {
                                     SingleModuleGraphBuilderNode::new_visited_module(target, *idx)
                                 } else {
-                                    SingleModuleGraphBuilderNode::new_module(target).await?
+                                    SingleModuleGraphBuilderNode::new_module(target, export).await?
```

Comment:
This is incorrect: in the first branch with `SingleModuleGraphBuilderNode::new_visited_module`, `export` is completely ignored.

At some point here during the module graph building, we need to merge the `ExportUsage`s  of multiple incoming edges of a single module to then get a single list of all `ExportUsage`s of some module.